src/games/strategy/triplea/ui/AbstractForumPosterPanel.java

```diff
@@ -21,13 +25,23 @@
   protected IPlayerBridge m_bridge;
   protected PBEMMessagePoster m_poster;
   protected TripleAFrame m_frame;
-  protected Action m_doneAction;
   protected ForumPosterComponent m_forumPosterComponent;
+  protected AbstractAction m_doneAction = new AbstractAction("Done") {
+    private static final long serialVersionUID = -5084680807785728744L;
+
+    @Override
+    public void actionPerformed(final ActionEvent e) {
+      if (m_forumPosterComponent.getHasPostedTurnSummary() || JOptionPane.YES_OPTION ==
```

Comment:
If we can avoid the `getHasPostedTurnSummary()` method call that would be good. Is it possible to initializes the m_doneAction in the constructor and know what this value will be without needing to get it from `m_forumPosterComponent`. Would you mind commenting on that and/or investigating?

The concern is that the initialization sequence of `m_doneAction`, `m_forumPosterComponent`, and then particularly `m_forumPosterDelegate`, are not very straight forward at all.